staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -772,20 +759,15 @@ func (p *sharedProcessor) removeListener(handle ResourceEventHandlerRegistration
 	return nil
 }
 
-func (p *sharedProcessor) distribute(obj interface{}, sync bool) {
+func (p *sharedProcessor) distribute(obj interface{}) {
 	p.listenersLock.RLock()
 	defer p.listenersLock.RUnlock()
 
-	for listener, isSyncing := range p.listeners {
```

Comment:
This doesn't appear to be a simple no-op cleanup/refactor to me.

Notably this PR makes it so `Update` events from a resync are now delivered to listeners that do not want sync events.

This line makes it possible to have listeners for which `isSyncing` may be true or false depending upon the last time we synced:

https://github.com/kubernetes/kubernetes/blob/88c8bcbb4aac388510adb1c7b1ed2b8a2f716996/staging/src/k8s.io/client-go/tools/cache/shared_informer.go#L832-L833

These lines determine the resync behavior by choosing the handler with the shorter period.

If you have two listeners: one with short resync period and one with resync disabled, will your change cause the listener with resync disabled to receive redundant sync events?

Reply:
Thank you for your response. I didn't notice the synchronization condition, and I appreciate you reminding me of that.  
I noticed that both types of messages are stored in the same list , and I thought it was unnecessary code. It was my lack of thorough consideration😰. 


Thank you for pointing out this design, thank you for help🙏.